### PR TITLE
bean-validation

### PR DESCRIPTION
**Pull request by girirajsharma**
Added comments(description) for bean validation annotations and added two new tests for bean validation in MemberValidationTest.java. Please ignore commit 5dee8a3.

### CODE REVIEW

File: bean-validation-custom-constraint/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+bean-validation-custom-constraints: Bean Validation using custom constraints via Arquillian Example
```
Comment (sgilda) [inline bean-validation-custom-constraint/README.md]: bean-validation-custom-constraint: Bean Validation using custom constraints via Arquillian Example

(The quickstart name must match the folder name)

File: bean-validation-custom-constraint/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+===================================================================================================
+Author: Giriraj Sharma  
+Level: Beginner  
+Technologies: Bean Validation, JPA  
+Summary: Shows how to use Arquillian to test Bean Validation using custom constraints
```
Comment (sgilda) [inline bean-validation-custom-constraint/README.md:6]: You need 2 spaces at the end of this line so it renders a carriage return.

File: bean-validation-custom-constraint/pom.xml [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>jboss-eap-bean-validation-customConstraint</groupId>
```
Comment (sgilda) [inline bean-validation-custom-constraint/pom.xml]: This should be:

```
    <groupId>org.jboss.quickstarts.eap</groupId>
```

File: bean-validation-custom-constraint/pom.xml [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+  <artifactId>jboss-eap-bean-validation-customConstraint</artifactId>
```
Comment (sgilda) [inline bean-validation-custom-constraint/pom.xml]: This should be:

```
    <artifactId>jboss-bean-validation-custom-constraint</artifactId>
```

File: bean-validation-custom-constraint/pom.xml [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+  <version>0.0.1-SNAPSHOT</version>
```
Comment (sgilda) [inline bean-validation-custom-constraint/pom.xml]:
```
    <version>6.2.0-redhat-SNAPSHOT</version>
```

File: bean-validation-custom-constraint/pom.xml [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+  <packaging>jar</packaging>
```
Comment (sgilda) [inline bean-validation-custom-constraint/pom.xml]: Doesn't this generate a WAR?

File: bean-validation-custom-constraint/pom.xml [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+
+  <name>JBoss EAP Quickstart: bean-validation-customConstraint</name>
```
Comment (sgilda) [inline bean-validation-custom-constraint/pom.xml]:
```
    <name>JBoss EAP Quickstart: bean-validation-custom-constraint</name>
```

File: bean-validation-custom-constraint/pom.xml [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+    <description>An Arquillian Bean Validation by Custom Constraints test example</description>
```
Comment (sgilda) [inline bean-validation-custom-constraint/pom.xml]:
```
    <description>An example that demonstrates Arquillian Bean Validation Custom Constraints </description>
```

File: bean-validation-custom-constraint/pom.xml [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+        <!-- [WARNING] Using platform encoding (UTF-8 actually) to copy filtered
+            resources, i.e. build is platform dependent! -->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <!-- JBoss dependency versions -->
+
+        <version.jboss.maven.plugin>7.4.Final</version.jboss.maven.plugin>
+
+        <!-- Define the version of the JBoss BOMs we want to import to specify tested stacks. -->
+        <version.jboss.bom.eap>6.2.1.GA</version.jboss.bom.eap>
```
Comment (sgilda) [inline bean-validation-custom-constraint/pom.xml]:
```
    <version.jboss.bom.eap>6.2.0-build-7</version.jboss.bom.eap>
```

File: bean-validation-custom-constraint/README.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+Target Product: EAP  
+Product Versions: EAP 6.1, EAP 6.2  
+Source: <https://github.com/jboss-developer/jboss-eap-quickstarts/>  
+
+What is it?
+-----------
+
+This project demonstrates how to use CDI 1.0, JPA 2.0 and Bean Validation 1.0. Bean Validation API allows the developers to define their own constraints by creating a new annotation and writing the validator which is used to validate the value. This quickstart will show you how to create custom constraints and 
+
+then use it to validate your data. It includes a persistence unit and some sample persistence code to introduce you to database access in enterprise Java. This quickstart does not contain a user interface layer.
```
Comment (sgilda) [inline bean-validation-custom-constraint/README.md:16]: I'm not sure why there is a blank line breaking up this sentence.

**Comment by sgilda** [comment]
Nice job on this!  

There are a few minor issues I discovered running the QS Tools. I noted them in the README and POM files. See step 8 under 'Basic Steps' in the Contributing Guide for details on how to run the QS Tools report. The problems with the POM file are also causing the Arquillian tests to fail.

After you fix the issues I mentioned, would you mind squashing the commits down to 1 commit? That makes it easier to review.

When you fix the issues, I will get someone to do a code review.

Thanks for submitting this quickstart!

**Comment by sgilda** [comment]
Even when I fix up your POM file, I am still getting an error when I run the tests. I see this in the server log:

13:26:26,152 ERROR [org.jboss.as.server](management-handler-thread - 5) JBAS015870: Deploy of deployment "test.war" was rolled back with the following failure message: 
{"JBAS014671: Failed services" => {"jboss.persistenceunit.\"test.war#primary\"" => "org.jboss.msc.service.StartException in service jboss.persistenceunit.\"test.war#primary\": javax.persistence.PersistenceException: [PersistenceUnit: primary] Unable to build EntityManagerFactory
    Caused by: javax.persistence.PersistenceException: [PersistenceUnit: primary] Unable to build EntityManagerFactory
    Caused by: org.hibernate.AnnotationException: No identifier specified for entity: org.jboss.as.quickstarts.bean_validation_customConstraint.MyPerson"}}

**Comment by girirajsharma** [comment]
@sgilda -Thanks a lot for reviewing and pointing out my errors.I shall try to fix all the bugs , you have mentioned. I will then perform testing using QS tools and will also see to it which problems in POM file are causing arquillian tests to fail. Once I fix the issues you mentioned, I would squash all the commits down to 1 commit so as to make it easy for code reviewer. Between,I also tried to extend the functionality of bean-validation quickstart(one which is already present in git master branch) in my commits (5dee8a3, a932f63, 341d79c, b1f898a, 9d93269, 7df11bb). Please review those commits also by a code reviewer whenever you feel like as per your convenience and time.Thanks.

File: bean-validation-custom-constraint/src/main/java/org/jboss/as/quickstarts/bean_validation_custom_constraint/AddressValidator.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+     * 4. The country in the address should be of atleast 4 characters.
+     */
+    public boolean isValid(MyAddress value, ConstraintValidatorContext context) {
+        if (value == null) {
+            return false;
+        }
+
+        if (value.getCity() == null || value.getCountry() == null || value.getLocality() == null
+            || value.getPinCode() == null || value.getState() == null || value.getStreetAddress() == null) {
+            return false;
```
Comment (LightGuard) [inline bean-validation-custom-constraint/src/main/java/org/jboss/as/quickstarts/bean_validation_custom_constraint/AddressValidator.java:40]: Shouldn't you test for empty too?

Comment (girirajsharma) [inline bean-validation-custom-constraint/src/main/java/org/jboss/as/quickstarts/bean_validation_custom_constraint/AddressValidator.java:40]: @LightGuard Thanks for pointing out. I shall make necessary changes and will review the code once again followed by testing.

File: bean-validation-custom-constraint/pom.xml (hunk 9 above)
Comment (girirajsharma) [inline bean-validation-custom-constraint/pom.xml]: @sgilda @LightGuard When I reset jboss.bom.eap version to 6.2.0-build-7, I face several build errors like  "Project build error: Non-resolvable import POM: Failure to find org.jboss.bom.eap:jboss-javaee-6.0-with-tools:pom:6.2.0-build-7 in http://maven.repository.redhat.com/earlyaccess/all/ was cached in the local repository, resolution will not be reattempted until the update interval of jboss-earlyaccess-repository has elapsed or updates are forced "  in dependency management.

**Comment by sgilda** [comment]
@girirajsharma : I just updated to get your latest changes and I get errors when I run `mvn clean test -Parq-jbossas-remote`

Here are the results:

Running org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest
Tests run: 5, Failures: 4, Errors: 0, Skipped: 0, Time elapsed: 3.231 sec <<< FAILURE!

Results :

Failed tests:   testAddressViolation(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): Address Field  was invalid expected:<Address Fields must not be null/empty and obey character limit constraints> but was:<null>
  testRegisterEmptyPerson(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): Three violations were found expected:<Name and Address fields must not be null/empty> but was:<null>
  testFirstNameSizeViolation(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): First Name was invalid expected:<size must be between 4 and 2147483647> but was:<3>
  testFirstNameNullViolation(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): First Name was invalid expected:<may not be null> but was:<null>

Tests run: 5, Failures: 4, Errors: 0, Skipped: 0

File: bean-validation-custom-constraint/README.md (hunk 10 above)
Comment (sgilda) [inline bean-validation-custom-constraint/README.md:16]: This shouldn't break between 'custom constraints and ' ... 'then use it to validate'

**Comment by sgilda** [comment]
I also noticed a few minor things in the README, but I can take care of them when everything else is complete.

Can you also squash the commits after you've made the fixes? Thanks!

**Comment by girirajsharma** [comment]
@sgilda The above logs actually indicate test failures in cases where either Person or Address entity beans are not set as per custom constraints. I will modify MyPersonTest.java by setting entity beans in such a way so that it doesn't gives any test failures and results in successful build.

**Comment by sgilda** [comment]
Closing this pull. @girirajsharma will issue a new pull.

**Pull request closed without merging**